examples/src/main/python/ml/quantile_discretizer_example.py

```diff
@@ -0,0 +1,40 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import print_function
+
+from pyspark import SparkContext
+from pyspark.sql import SQLContext
+# $example on$
+from pyspark.ml.feature import QuantileDiscretizer
+# $example off$
+
+if __name__ == "__main__":
+    sc = SparkContext(appName="QuantileDiscretizerExample")
```

Comment:
Examples have been updated to use SparkSession, see e.g. https://github.com/apache/spark/blob/master/examples/src/main/python/ml/binarizer_example.py. 

Also, there is a PR to indent the builder pattern #12911 so you should follow that style (also refer #12809)